scripts/accessibility-check.js

```diff
@@ -1,221 +1,94 @@
+    // Get the test ids from storybook
+    const testIds = await getUrls(browser);
 
-    const untriggeredExpectedIssues = Object.entries(expectedIssues);
-    const untriggeredExpectedViolationsCount = untriggeredExpectedIssues.length;
+    console.log(chalk.bold(`🧪 Testing ${testIds.length} urls with axe`));
 
-    console.log(
-      `There were ${totalViolationsCount} Issues reported! ${expectedViolationsCount} Issues were expected. ${untriggeredExpectedViolationsCount} Expected Issues were absent`,
-    );
+    // Test the pages with axe
+    const testPage = async (url) => {
+      const id = url.replace(`${iframePath}?id=all-components-`, '');
+      console.log(` - ${id}`);
 
-    if (
-      unexpectedViolationsCount === 0 &&
-      untriggeredExpectedViolationsCount === 0 &&
-      errorCount === 0
-    ) {
-      return;
-    }
+      const page = await browser.newPage();
+      await page.goto(url);
+      const result = await page.evaluate(() => window.axe.run(document.getElementById('root'), {}));
+      await page.close();
 
-    if (unexpectedViolationsCount) {
-      console.log('---\n\nUnexpected Issues:');
-      resultsWithUnexpectedViolations.forEach((result) => {
-        console.log(
-          `${result.type} ${result.url} (${result.errorCount}): \n${result.error}`,
-        );
-      });
+      if (result.violations.length) {
+        return formatMessage(id, result.violations);
+      }
     }
 
-    if (untriggeredExpectedViolationsCount) {
-      console.log('---\n\nExpected Issues that were not triggerd:');
-      untriggeredExpectedIssues.forEach(([url, expectedViolationCount]) => {
-        const result = results.find((resultItem) => resultItem.url === url);
+    const results = await pMap(testIds, testPage, {concurrency: concurrentCount});
+    await browser.close();
 
-        if (!result) {
-          console.log(
-            `${url}: No matching story, remove this url from the expectedIssues array`,
-          );
-        } else if (result.type !== 'ERROR') {
-          console.log(
-            `${url}: Expected ${expectedViolationCount} issues, got ${result.errorCount}.`,
-          );
-        }
-      });
+    // Format the results and log them out
+    const messages = results.filter(x => x);
+    if(messages.length) {
+      console.error(chalk.bold(`‼️  Accessibility failures found`));
+      console.log(messages.join('\n'));
+      process.exit(1);
     }
-
-    if (errorCount.length) {
-      console.log(
-        '---\n\nPages triggered an ERROR when trying to identify violations (you should rerun too see if this goes away):',
-      );
-
-      resultsWithErrors.forEach((result) => {
-        console.log(`${result.type} ${result.url}: \n${result.error}`);
-      });
+    else {
+      console.log(chalk.bold('✅ Accessibility is :chefkiss:'));
     }
-
-    process.exit(1);
-  } catch (err) {
-    console.log(err);
+  } catch (error) {
+    console.error(error.message);
     process.exit(1);
```

Comment:
`process.exit(1);` is causing error responses from CI

Reply:
Presumably if there is an error when executing the script we'd want it the job to fail? Otherwise how would we notice that it isn't working?

Reply:
This didn't do a good job at highlighting the problem. No matter what happens it ends with `process.exit(1)`.

https://github.com/Shopify/polaris-react/blob/a334af33d0c9a16babedb1d8ec9efd9a96690035/scripts/accessibility-check.js#L206-L220

Reply:
If there are no errors then it bails out early here: https://github.com/Shopify/polaris-react/blob/a334af33d0c9a16babedb1d8ec9efd9a96690035/scripts/accessibility-check.js#L172-L178